addons/mail/static/src/new/thread/message_seen_indidator.js

```diff
@@ -0,0 +1,47 @@
+/* @odoo-module */
+
+import { Component } from "@odoo/owl";
+import { useStore } from "../core/messaging_hook";
+
+export class MessageSeenIndicator extends Component {
+    static template = "mail.message_seen_indicator";
+    static components = {};
+    static props = ["message", "thread"];
+    static defaultProps = {};
+
+    setup() {
+        this.store = useStore();
+    }
+
+    get hasEveryoneSeen() {
+        const otherPartnerSeenInfosDidNotSee = this.props.thread.partnerSeenInfos.filter(
+            (partnerSeenInfo) =>
+                partnerSeenInfo.partner.id !== this.props.message.author.id &&
+                (!partnerSeenInfo.lastSeenMessage ||
+                    partnerSeenInfo.lastSeenMessage.id < this.props.message.id)
+        );
+        return otherPartnerSeenInfosDidNotSee.length === 0;
+    }
+
+    get isMessagePreviousToLastCurrentPartnerMessageSeenByEveryone() {
+        return this.props.message.id < this.props.thread.lastCurrentPartnerMessageSeenByEveryone.id;
+    }
+
+    get hasSomeoneSeen() {
+        const otherPartnerSeenInfosSeen = this.props.thread.partnerSeenInfos.filter(
+            (partnerSeenInfo) =>
+                partnerSeenInfo.partner.id !== this.props.message.author.id &&
+                partnerSeenInfo.lastSeenMessage?.id >= this.props.message.id
```

Comment:
partnerSeenInfo.lastSeenMessage &&
                        partnerSeenInfo.lastSeenMessage.id >= this.message.id